### ISSUE TITLE
Adding a keyvault feature in  azure datafactory linked service

### ISSUE THREAD

**Issue by murali1994** [terraform-providers/terraform-provider-azurerm#6481]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I have found some configuration is missing in datafactory_linkedservice .As in terraform only connection string variable is present but in azure you can see two options one is connection string and other one is AKV so we as a team decided there should be AKV(Azure keyvault) variable embedded in the module.Because adding only connection string actually gives less value because all the users who have access to datafactory can have ability to get the connection string .So need the community help to add this feature .

### New or Affected Resource(s)

azurerm_data_factory_linked_service_sql_server

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by pearcec** [comment]
Looks like a duplicate of #6264

**Comment by sebastianreloaded** [comment]
copying my comment here:
The Azure Rest API declares a special property typeProperties.password additionally to typeProperties.connectionString, which accepts AzureKeyVaultSecretReference. It would be a great improvement for security using this setup.

See https://docs.microsoft.com/en-us/rest/api/datafactory/linkedservices/createorupdate#sqlserverlinkedservice

**Comment by lfraile** [comment]
Hello, is there any advance on this issue?

Thank you

**Comment by allantargino** [comment]
Hey, I'm adding this feature on the new synapse linked service (#9928). Once this is approved and merged, I can work on SQL Server linked service as well and add suport for key vault.

**Comment by lfraile** [comment]
Seems great, I would be happy to help, but as today, my knowledge of Go is: 0 

But if you need me to test any previous version, just let me know @allantargino
